runners/dmn-tck-runner-drools/src/test/java/org/omg/dmn/tck/runner/drools/DroolsTCKTest.java

```diff
@@ -57,9 +71,49 @@
    implements DmnTckVendorTestSuite
 {
 
+   private static final SimpleFileVisitor<Path> rmrf = new SimpleFileVisitor<Path>() {
+      @Override
+      public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+         Files.delete(file);
+         return FileVisitResult.CONTINUE;
+      }
+
+      @Override
+      public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+         Files.delete(dir);
+         return FileVisitResult.CONTINUE;
+      }
+   };
    private static final Logger logger = LoggerFactory.getLogger(DroolsTCKTest.class);
    public static final BigDecimal NUMBER_COMPARISON_PRECISION = new BigDecimal("0.00000001");
 
+   File propsFile;
+   Properties props;
+
+   {
```

Comment:
Just a general question, instead of using a static class context for File management, couldn't there be a solution where this code is used during test suite method calls?

Reply:
I am not sure what you suggest and why we would want to do it during suite method calls. Can you maybe make an example?

Reply:
What I'm trying to say is that this block of code is performing a lot of file system IO in the class static context, and there are many thing which could go wrong. By the way is strucutred now, instead of failing "gracefully", it is simply crashing at init, like the following (simulated locally directory structure in results problem):
```
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ tck-runner-drools ---
[INFO] Surefire report directory: /home/mmortari/git/GH/dmn-tck/dmn-tck/runners/dmn-tck-runner-drools/target/surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.omg.dmn.tck.runner.drools.DroolsTCKTest
19:51:52.464 [main] ERROR o.o.d.tck.runner.junit4.DmnTckSuite.<init>:53 - Error instantiating test suite.
java.lang.ArrayIndexOutOfBoundsException: -1
	at org.omg.dmn.tck.runner.drools.DroolsTCKTest.<init>(DroolsTCKTest.java:98) ~[test-classes/:na]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) [na:1.8.0_171]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) [na:1.8.0_171]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) [na:1.8.0_171]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) [na:1.8.0_171]
	at java.lang.Class.newInstance(Class.java:442) ~[na:1.8.0_171]
	at org.omg.dmn.tck.runner.junit4.DmnTckSuite.<init>(DmnTckSuite.java:51) ~[tck-runner-1.0.0-SNAPSHOT.jar:na]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) [na:1.8.0_171]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) [na:1.8.0_171]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) [na:1.8.0_171]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) [na:1.8.0_171]
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104) [junit-4.12.jar:4.12]
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86) [junit-4.12.jar:4.12]
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59) [junit-4.12.jar:4.12]
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26) [junit-4.12.jar:4.12]
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59) [junit-4.12.jar:4.12]
	at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33) [junit-4.12.jar:4.12]
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:250) [surefire-junit4-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141) [surefire-junit4-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112) [surefire-junit4-2.12.4.jar:2.12.4]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_171]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_171]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_171]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_171]
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189) [surefire-api-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165) [surefire-booter-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85) [surefire-booter-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115) [surefire-booter-2.12.4.jar:2.12.4]
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75) [surefire-booter-2.12.4.jar:2.12.4]
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.186 sec <<< FAILURE!
initializationError(org.omg.dmn.tck.runner.drools.DroolsTCKTest)  Time elapsed: 0.002 sec  <<< ERROR!
java.lang.ArrayIndexOutOfBoundsException: -1
	at org.omg.dmn.tck.runner.drools.DroolsTCKTest.<init>(DroolsTCKTest.java:98)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at org.omg.dmn.tck.runner.junit4.DmnTckSuite.<init>(DmnTckSuite.java:51)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:250)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)


Results :

Tests in error: 
  initializationError(org.omg.dmn.tck.runner.drools.DroolsTCKTest): -1

Tests run: 1, Failures: 0, Errors: 1, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] DMN :: TCK :: Marshallers .......................... SUCCESS [  2.674 s]
[INFO] DMN :: TCK :: jUnit Runner ......................... SUCCESS [  0.590 s]
[INFO] DMN :: TCK :: Report Generator ..................... SUCCESS [  0.312 s]
[INFO] DMN :: TCK :: jUnit Runner :: Drools ............... FAILURE [  0.687 s]
[INFO] DMN :: TCK :: Runners .............................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

Now my point being, the read of the Property (to update the "last update") is not really mandatory at least to run the test suite and display those results to a TCK user.

See for instance here: https://github.com/dmn-tck/tck/blob/56cc6640428fb0dee4815cd79f0bb3b5ba1d044d/runners/dmn-tck-runner-drools/src/test/java/org/omg/dmn/tck/runner/drools/DroolsTCKTest.java#L78
Could be an analogous case.

Hence my suggestion was to move this block under say for example method `createContext()` which looks the proper for initialization of the whole suite, and gracefully handle the case whereby any filesystem IO problem is not making the whole thing to crash.

What do you think?

p.s.: I have found other details which I will describe in other comments to the appropriate lines